File: core/src/main/java/hello/core/OrderApp.java
package hello.core;

import hello.core.member.data.Grade;
import hello.core.member.data.Member;
import hello.core.member.service.MemberService;
import hello.core.member.service.MemberServiceImpl;
import hello.core.order.data.Order;
import hello.core.order.implement.OrderServiceImpl;
import hello.core.order.service.OrderService;

public class OrderApp {

    public static void main(String[] args) {
        AppConfig appConfig = new AppConfig();

        MemberService memberService = appConfig.memberService();
        OrderService orderService = appConfig.orderService();

        Long memberId = 1L;
        Member member = new Member(memberId, "memberA", Grade.VIP);
        memberService.join(member);
        Order order = orderService.createOrder(memberId, "itemA", 10000);

        System.out.println("order = " + order.toString());
        System.out.println("Calculated Price = " + order.calculatePrice());
    }

}

File: core-kotlin/src/main/kotlin/hello/corekotlin/OrderApp.kt
package hello.corekotlin


import hello.corekotlin.member.data.Grade
import hello.corekotlin.member.data.Member
import hello.corekotlin.member.service.MemberService
import hello.corekotlin.member.service.MemberServiceImpl
import hello.corekotlin.order.implement.OrderServiceImpl
import hello.corekotlin.order.service.OrderService
import org.springframework.context.annotation.AnnotationConfigApplicationContext


fun main() {

//    val appConfig = AppConfig()
//
//    val memberService = appConfig.memberService()
//    val orderService = appConfig.orderService()

    val applicationContext = AnnotationConfigApplicationContext(AppConfig::class.java)
    val memberService = applicationContext.getBean("memberService", MemberService::class.java)
    val orderService = applicationContext.getBean("orderService", OrderService::class.java)

    val memberId = 1L
    val member = Member(
        memberId,
        "memberA",
        Grade.VIP

    )


    memberService.join(member)
    val order = orderService.createOrder(memberId, "itemA", 20000)

    println("order = " + order.toString())
    println("discountedPrice = " + order.calculatePrice().toString())
}